### ISSUE TITLE
Goto definition throws deprecation/warnings on 7.4

### ISSUE THREAD

**Issue by dantleech** [phpactor/phpactor#861]
```
E474: Unidentified byte: PHP Deprecated:  Unparenthesized `a ? b : c ?: d` is deprecated. Use either `(a ? b : c) ?: d` or `a ? b : (c ?
: d)` in /home/daniel/.vim/plugged/phpactor/vendor/phpactor/worse-reflection/lib/Core/Type.php on line 202^@{"version":"1.0.0","action":
"open_file","parameters":{"path":"\/home\/daniel\/www\/phpactor\/phpactor\/vendor\/phpactor\/code-builder\/lib\/Adapter\/TolerantParser\
/TolerantUpdater.php","offset":1313,"force_reload":false,"target":"focused_window"}}
E474: Failed to parse PHP Deprecated:  Unparenthesized `a ? b : c ?: d` is deprecated. Use either `(a ? b : c) ?: d` or `a ? b : (c ?: d
)` in /home/daniel/.vim/plugged/phpactor/vendor/phpactor/worse-reflection/lib/Core/Type.php on line 202^@{"version":"1.0.0","action":"op
en_file","parameters":{"path":"\/home\/daniel\/www\/phpactor\/phpactor\/vendor\/phpactor\/code-builder\/lib\/Adapter\/TolerantParser\/To
lerantUpdater.php","offset":1313,"force_reload":false,"target":"focused_window"}}
Error detected while processing function phpactor#GotoDefinition[1]..phpactor#_GotoDefinitionTarget[1]..phpactor#rpc[15]..phpactor#_rpc_
dispatch
```

**Comment by elythyr** [comment]
Can you add the deprecation warning in the issue ?

**Comment by dantleech** [comment]
Done :+1:

**Comment by elythyr** [comment]
I don't get it, the only line with a ternary in master is:
https://github.com/phpactor/worse-reflection/blob/d72346cedf13452f59f87b24e6e24b4f89315dda/lib/Core/Type.php#L211

But the parenthesis are correct according to https://www.php.net/manual/en/migration74.deprecated.php

Could it came from a local version of `worse-reflection` on your computer ?

**Comment by gabrielbidula** [comment]
Hi guys, i got the same issue when trying to go to definitition, using php 7.4

`Error detected while processing function phpactor#GotoDefinition[1]..phpactor#_GotoDefinitionTarget[1]..phpactor#rpc:
line   10:
E474: Unidentified byte: PHP Deprecated:  Unparenthesized `a ? b : c ?: d` is deprecated. Use either `(a ? b : c) ?: d` or `a ? b : (c ?: d)` in /Users/Gabriel/.config/nvim/plugged/phpactor/vendor/phpactor/worse-reflection/lib/Core/Type
.php on line 202^@^@Deprecated: Unparenthesized `a ? b : c ?: d` is deprecated. Use either `(a ? b : c) ?: d` or `a ? b : (c ?: d)` in /Users/Gabriel/.config/nvim/plugged/phpactor/vendor/phpactor/worse-reflection/lib/Core/Type.php on li
ne 202^@{"version":"1.0.0","action":"open_file","parameters":{"path":"\/Users\/Gabriel\/Dev\/vsen.gg\/api\/src\/Entity\/SearchMatch.php","offset":197,"force_reload":false,"target":"focused_window"}}
E474: Failed to parse PHP Deprecated:  Unparenthesized `a ? b : c ?: d` is deprecated. Use either `(a ? b : c) ?: d` or `a ? b : (c ?: d)` in /Users/Gabriel/.config/nvim/plugged/phpactor/vendor/phpactor/worse-reflection/lib/Core/Type.ph
p on line 202^@^@Deprecated: Unparenthesized `a ? b : c ?: d` is deprecated. Use either `(a ? b : c) ?: d` or `a ? b : (c ?: d)` in /Users/Gabriel/.config/nvim/plugged/phpactor/vendor/phpactor/worse-reflection/lib/Core/Type.php on line
202^@{"version":"1.0.0","action":"open_file","parameters":{"path":"\/Users\/Gabriel\/Dev\/vsen.gg\/api\/src\/Entity\/SearchMatch.php","offset":197,"force_reload":false,"target":"focused_window"}}
Error detected while processing function phpactor#GotoDefinition[1]..phpactor#_GotoDefinitionTarget[1]..phpactor#rpc[15]..phpactor#_rpc_dispatch:
line  194:
E605: Exception not caught: Do not know how to handle action '0'
Error detected while processing function phpactor#GotoDefinition[1]..phpactor#_GotoDefinitionTarget[1]..phpactor#rpc:
line   15:
E171: Missing :endif`

**Comment by rdamborsky** [comment]
Thanks @gabrielbidula, this worked for me

**Comment by elythyr** [comment]
This is fix on master but the last tag `0.3.1` was created in august.
I think we should create a minor tag to remove this deprecation.
And maybe also make all dependencies on `develop` branch to the `dev-master` version so that people on `develop` actual get the latest updates ?

**Comment by dantleech** [comment]
Closing as this is not an issue in develop will be fixed in the following release.
